extensions/amp-video/0.1/amp-video.js

```diff
@@ -765,7 +765,18 @@ export class AmpVideo extends AMP.BaseElement {
    */
   setUpCaptions_() {
     const captionsId = this.element.getAttribute('captions-id');
-    if (!captionsId) {
+    // if (!captionsId) {
+    //   return;
+    // }
+
+    const trackEl = this.element.querySelector('track');
+    if (!captionsId && trackEl) {
+      // Works because this function is called twice, so this function runs again after it is attached from the first time it's called.
+      const captionsId = trackEl.getAttribute('src');
+      const captionsEl = this.win.document.createElement('amp-story-captions');
+      captionsEl.setAttribute('id', captionsId);
+      this.element.setAttribute('captions-id', captionsId);
+      this.element.appendChild(captionsEl);
```

Comment:
Please move this block to `video-cache.js` that is responsible for setting sources/tracks provided by the Google Cache. It should still be the one place that configures whatever the Google video cache provides.

Reply:
My bad this is WIP code meant for a different PR. 
I'm saving these comments and moving this into another PR.